src/pages/search/search.tsx

```diff
@@ -1,82 +1,76 @@
+import { SearchControls } from '../../components/search-controls/search-controls.tsx';
 
-interface ISearchState {
-  charList: Character[] | undefined;
-  query: string;
-  loading: boolean;
-}
+const Search: FC = () => {
+  const [charList, setCharList] = useState<Character[]>([]);
+  const [isLoading, setLoading] = useState(false);
+  const [showControls, setShowControls] = useState(false);
+  const { savedQuery, setQueryLS } = useLocalStorage(LOCAL_STORAGE_KEY);
+  const { page, resetPage, maxPage, setMaxPage, prevPage, nextPage } =
+    usePagination();
+  useQueryUpdate(page, savedQuery);
 
-class Search extends Component {
-  state: ISearchState = {
-    query: '',
-    charList: undefined,
-    loading: false,
-  };
+  const handleClick = useCallback(
+    (name: string, queryPage: number) => {
+      setShowControls(false);
+      setLoading(true);
+      getCharacters(name, queryPage)
+        .then((response) => {
+          setCharList(response.characters);
+          setMaxPage(response.maxPage);
+          setShowControls(true);
+        })
+        .catch(() => {
+          setCharList([]);
+          setShowControls(false);
+        })
+        .finally(() => setLoading(false));
+    },
+    [setMaxPage]
+  );
 
-  componentDidMount(): void {
-    this.setState((prevState) => ({
-      ...prevState,
-      query: localStorage.getItem('search-query') || '',
-    }));
-    this.handleClick();
-  }
+  useEffect(() => {
+    handleClick(savedQuery, page);
+  }, [handleClick, savedQuery, page]);
 
-  handleClick = (query = this.state.query): void => {
-    this.setState({ loading: true });
-    getCharacters(query)
-      .then((newCharList) => {
-        this.setState({ charList: newCharList });
-      })
-      .catch(() => {
-        this.setState({ charList: [] });
-      })
-      .finally(() => this.setState({ loading: false }));
-  };
-
-  setQuery = (query: string): void => {
-    this.setState({ query });
-    localStorage.setItem('search-query', query);
-  };
-
-  render() {
-    return (
-      <div
-        className={style.search}
-        style={{ background: `url(${bgPath}) center center/cover no-repeat` }}
-      >
-        <h1 className={style['search-header']}>Rick and Morty</h1>
-        <span className={style['search-subheader']}>characters database</span>
-        <SearchForm
-          query={this.state.query}
-          setQuery={this.setQuery}
-          clickFn={this.handleClick}
-        />
-        <div className={style['search-results']}>
-          {this.state.loading ? (
-            <Loading />
-          ) : (
-            <ErrorBoundary>
-              <CardList charList={this.state.charList} />
-            </ErrorBoundary>
-          )}
-        </div>
-        <Button
-          callback={() => this.handleClick('qwe213')}
-          text="Error"
-          className={style['error-button']}
-        />
+  return (
+    <div className={style.search}>
+      <SearchForm
+        clickFn={handleClick}
```

Comment:
not a very good naming for prop and callback
```suggestion
        onSearch={handleSearch}
```

Reply:
Thank. Unfortunately, I'm not very good at naming variables and functions.